### PR TITLE
[SPARK-34749][SQL] Simplify ResolveCreateNamedStruct

### PR DESCRIPTION
**Pull request by cloud-fan**
### What changes were proposed in this pull request?

This is a follow-up of https://github.com/apache/spark/pull/31808 and simplifies its fix to one line (excluding comments).

### Why are the changes needed?

code simplification

### Does this PR introduce _any_ user-facing change?

no

### How was this patch tested?

N/A

### CODE REVIEW

**Comment by cloud-fan** [comment]
cc @dongjoon-hyun @maropu

**Comment by dongjoon-hyun** [comment]
Could you fix the relevant UT failure?
```
[info] - dereference *** FAILED *** (335 milliseconds)
[info]   == FAIL: Plans do not match ===
[info]   !'Filter struct(a, 'a, b, 'b)[b]   'Filter named_struct(NamePlaceholder, 'a, NamePlaceholder, 'b)[b]
[info]    +- OneRowRelation                 +- OneRowRelation (PlanTest.scala:161)
```

**Comment by cloud-fan** [comment]
The scala 2.13 failure is unrelated. Can you take another look? @dongjoon-hyun @maropu , thanks!

**Comment by LuciferYang** [comment]
> The scala 2.13 failure is unrelated. Can you take another look? @dongjoon-hyun @maropu , thanks!


https://github.com/apache/spark/pull/31862 seems can fix this.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/complexTypeCreator.scala [hunk 1]
```diff
@@ -339,6 +339,14 @@ object CreateStruct {
...
   def apply(children: Seq[Expression]): CreateNamedStruct = {
     CreateNamedStruct(children.zipWithIndex.flatMap {
+      // For multi-part column name like `struct(a.b.c)`, it may be resolved into:
+      //   1. Attribute if `a.b.c` is simply a qualified column name.
+      //   2. GetStructField if `a.b` refers to a struct-type column.
+      //   3. GetArrayStructFields if `a.b` refers to a array-of-struct-type column.
+      //   4. GetMapValue if `a.b` refers to a map-type column.
+      // We should always use the last part of the column name (`c` in the above example) as the
+      // alias name inside CreateNamedStruct.
+      case (u: UnresolvedAttribute, _) => Seq(Literal(u.nameParts.last), u)
```
Comment (maropu) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/complexTypeCreator.scala:349]: Ah, I see. This change makes `ResolveCreateNamedStruct` simpler. The fix seems smart.

**Review by maropu — APPROVED** [review]

**Comment by maropu** [comment]
Thanks! Merged to master.

**Comment by maropu** [comment]
The commit has a conflict with branch-3.1, so could you open a backport PR for the fix?

**Pull request closed without merging**